engine/source/runtime/function/render/source/vulkan_manager/context/vulkan_context.cpp

```diff
@@ -323,11 +323,33 @@ void Pilot::PVulkanContext::initializePhysicalDevice()
         std::vector<VkPhysicalDevice> physical_devices(physical_device_count);
         vkEnumeratePhysicalDevices(_instance, &physical_device_count, physical_devices.data());
 
+        std::vector<std::pair<int, VkPhysicalDevice>> ranked_physical_devices;
         for (const auto& device : physical_devices)
         {
-            if (isDeviceSuitable(device))
+            VkPhysicalDeviceProperties physical_device_properties;
+            vkGetPhysicalDeviceProperties(device, &physical_device_properties);
+            int score = 0;
+
+            if (physical_device_properties.deviceType == VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU)
```

Comment:
Suggest following the algorithm of [VulkanTutorial](https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Physical_devices_and_queue_families), not just considering the device type, but also including the limits, features

Reply:
Ok I see taht. In my previous project (doing that while learning with Vulkan Tutorial) I will enumrate all the features to score gpu, I don't know if that's better.